numba_dpex/experimental/models.py

```diff
@@ -2,19 +2,39 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
-"""Provides Numba datamodel for the numba_dpex types introduced in the
+"""Provides the Numba data models for the numba_dpex types introduced in the
 numba_dpex.experimental module.
 """
 
-from numba.core.datamodel import models
+from numba.core.datamodel import DataModelManager, models
 from numba.core.extending import register_model
 
-from numba_dpex.core.datamodel.models import dpex_data_model_manager as dmm
+import numba_dpex.core.datamodel.models as dpex_core_models
 
 from .types import KernelDispatcherType
 
-# Register the types and datamodel in the DpexKernelTargetContext
-dmm.register(KernelDispatcherType, models.OpaqueModel)
 
-# Register the types and datamodel in the DpexTargetContext
+def _init_exp_data_model_manager() -> DataModelManager:
+    """Initializes a DpexExpKernelTarget-specific data model manager.
+
+    Extends the DpexKernelTargetContext's datamodel manager with all
+    experimental types that are getting added to the kernel API.
+
+    Returns:
+        DataModelManager: A numba-dpex DpexExpKernelTarget-specific data model
+        manager
+    """
+
+    dmm = dpex_core_models.dpex_data_model_manager.copy()
+
+    return dmm
+
+
+exp_dmm = _init_exp_data_model_manager()
+
+
+# Register the types and data model in the DpexExpTargetContext
+# Add here...
+
+# Register any new type that should go into numba.core.datamodel.default_manager
 register_model(KernelDispatcherType)(models.OpaqueModel)
```

Comment:
May be it should be registered in experimental data model?

Reply:
Well we want it available inside `dpjit` that uses Numba's default data model manager. To properly do it, we will have to create a separate data model manager for dpjit target. Maybe separate PR?